animation_nodes/nodes/color/set_vertex_color.py

```diff
@@ -1,62 +1,101 @@
 import bpy
-import numpy
 from bpy.props import *
 from ... events import propertyChanged
-from ... base_types import AnimationNode
+from ... base_types import AnimationNode, VectorizedSocket
+from ... data_structures import Color, ColorList, VirtualColorList
+from . c_utils import getLoopColorsFromVertexColors, getLoopColorsFromPolygonColors
+
+colorModeItems = [
+    ("LOOP", "Loop", "Set color of every loop vertex", "NONE", 0),
+    ("VERTEX", "Vertex", "Set color of every vertex", "NONE", 1),
+    ("POLYGON", "Polygon", "Set color of every polgon", "NONE", 2)    
+]
+
+colorLayerIdentifierTypeItems = [
+    ("INDEX", "Index", "Get vertex color layer based on the index", "NONE", 0),
+    ("NAME", "Name", "Get vertex color layer based on the name", "NONE", 1)
+]
 
 class SetVertexColorNode(bpy.types.Node, AnimationNode):
     bl_idname = "an_SetVertexColorNode"
     bl_label = "Set Vertex Color"
     errorHandlingType = "EXCEPTION"
 
-    vertexColorName: StringProperty(name = "Vertex Color Group", default = "Col", update = propertyChanged)
-    checkIfColorIsSet: BoolProperty(default = True)
+    colorMode: EnumProperty(name = "Color Mode", default = "LOOP",
+        items = colorModeItems, update = AnimationNode.refresh)
```

Comment:
The update function should be `propertyChanged`. `AnimationNode.refresh` will trigger a redraw of the node, which is not needed in this case.

Reply:
Okay. Thanks for the info :) It is changed.